monai/handlers/roc_auc.py

```diff
@@ -17,8 +17,8 @@
 from monai.metrics import compute_roc_auc
 from monai.utils import Average, exact_version, optional_import
 
-idist, _ = optional_import("ignite", "0.4.2", exact_version, "distributed")
-EpochMetric, _ = optional_import("ignite.metrics", "0.4.2", exact_version, "EpochMetric")
+idist, _ = optional_import("ignite", "0.4.4", exact_version, "distributed")
+EpochMetric, _ = optional_import("ignite.metrics", "0.4.4", exact_version, "EpochMetric")
 
 
 class ROCAUC(EpochMetric):  # type: ignore[valid-type, misc]  # due to optional_import
```

Comment:
@Nic-Ma do you think it would be more interesting to improve and adapt our ROCAUC implementation or this one is very specific for your use-cases ?

Reply:
Hi @vfdev-5 ,

There are 3 reasons that I didn't use the AUC implementation in ignite:
1. The ignite version is based on `sklearn.metrics`, we don't want to involve this dependency, it's self-contained here.
2. We need to support the case that predictions at different ranks have different lengths for distributed evaluation:
https://github.com/Project-MONAI/MONAI/blob/master/monai/handlers/roc_auc.py#L91
3. The AUC metric of ignite is in contrib folder, we don't want to involve your contrib code so far.

Thanks.